test/widgets/lightbox_test.dart

```diff
@@ -289,6 +308,84 @@ void main() {
     //   https://github.com/zulip/zulip-flutter/pull/833#issuecomment-2251782337
   });
 
+  group('IconButton Download Test', () {
+    Future<void> buildTestWidget(WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Builder(
+              builder: (context) {
+                return IconButton(
+                  icon: const Icon(Icons.download_rounded),
+                  onPressed: () async {
+                    // Simulate download logic
+                    final response = await fakeHttpClient.get(Uri.parse('https://example.com/image.png'));
+
+                    if (response.statusCode == 200) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(content: Text("Image downloaded successfully")),
+                      );
+                    } else {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(content: Text("Failed to download image")),
+                      );
+                    }
+                  },
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
```

Comment:
Generally we should use `TestZulipApp` for testing, unless we need `ZulipApp` specifically. You can see other test cases for example.